platform/podjs/src/browserPod.ts

```diff
@@ -15,68 +15,130 @@ import * as zip from "@zip.js/zip.js";
 import endpointsJson from "../../../../polyPod-config/endpoints.json";
 import { Manifest, readManifest } from "./manifest";
 
+const DB_NAME = "polypod";
+const DB_VERSION = 1;
+const OBJECT_STORE_POLY_IN = "poly-in";
+const OBJECT_STORE_POLY_OUT = "poly-out";
+
 const NAV_FRAME_ID = "polyNavFrame";
 const NAV_DEFAULT_BACKGROUND_COLOR = "#ffffff";
 const NAV_DARK_FOREGROUND_COLOR = "#000000";
 const NAV_LIGHT_FOREGROUND_COLOR = "#ffffff";
 
-class LocalStoragePolyIn implements PolyIn {
-    private static readonly storageKey = "polyInStore";
-    private store = JSON.parse(
-        localStorage.getItem(LocalStoragePolyIn.storageKey) || "[]"
-    );
+async function migrateData(db: IDBDatabase): Promise<IDBDatabase> {
+    const polyInData = localStorage.getItem("polyInStore");
+    const polyOutData = localStorage.getItem("files");
+
+    if (polyInData || polyOutData) {
+        const quads = polyInData ? JSON.parse(polyInData) : [];
+        const files: FileInfo[] = [];
+
+        if (polyOutData) {
+            for (const [, file] of JSON.parse(polyOutData)) {
+                const dataUrl = localStorage.getItem(file.id);
+                if (dataUrl) {
+                    files.push({
+                        id: file.id,
+                        name: file.name,
+                        time: file.time,
+                        blob: await (await fetch(dataUrl)).blob(),
+                    });
+                }
+            }
+        }
 
-    async match(matcher: Partial<Matcher>): Promise<RDF.Quad[]> {
-        return this.store.filter((quad: RDF.Quad) => {
-            if (matcher.subject && quad.subject.value != matcher.subject.value)
-                return false;
-            if (matcher.object && quad.object.value != matcher.object.value)
-                return false;
-            if (
-                matcher.predicate &&
-                quad.predicate.value != matcher.predicate.value
-            )
-                return false;
-            return true;
+        await new Promise((resolve, reject) => {
```

Comment:
If I'm understanding this correctly, you're extracting the information from localStorage and putting it into IndexedDB. So far, so good. But if all information needs to be initially in localStorage, will we not have the same limit as before?

Reply:
It's just a migration - if there is any data there, we'll keep it rather than throwing it away. Any data that was there before was subject to the localStorage limitations, yes, but any data that is there is valid - if it would have been too big, it wouldn't have been saved..

Reply:
There is no requirement to have anything in `localStorage` initially. As @fhd explained, we are merely carrying over any data that may have been stored in `localStorage` during previous sessions.

Reply:
I'd be stumped if there was any meaningful data there.